raphtory/src/db/graph/views/filter/node_type_filtered_graph.rs

```diff
@@ -168,90 +168,1338 @@ mod tests_node_type_filtered_subgraph {
             prelude::{AdditionOps, NodeViewOps},
         };
 
-        macro_rules! assert_filter_results {
-            ($filter_fn:ident, $filter:expr, $node_types:expr, $expected_results:expr) => {{
-                let filter_results = $filter_fn($filter.clone(), $node_types.clone());
-                assert_eq!($expected_results, filter_results);
+        macro_rules! assert_filter_nodes_results {
+            // Default usage with all variants
+            ($init_fn:ident, $filter:expr, $node_types:expr, $expected:expr) => {
+                assert_filter_nodes_results!(
+                    $init_fn,
+                    $filter,
+                    $node_types,
+                    $expected,
+                    variants = [graph, persistent_graph, event_disk_graph, persistent_disk_graph]
+                );
+            };
+
+            // Custom variant list
+            ($init_fn:ident, $filter:expr, $node_types:expr, $expected:expr, variants = [$($variant:ident),+ $(,)?]) => {{
+                $(
+                    assert_filter_nodes_results_variant!($init_fn, $filter.clone(), $node_types.clone(), $expected, $variant);
+                )+
+            }};
+        }
+
+        macro_rules! assert_filter_nodes_results_variant {
+            ($init_fn:ident, $filter:expr, $node_types:expr, $expected:expr, graph) => {{
+                let graph = $init_fn(Graph::new());
+                let node_types: Vec<String> =
```

Comment:
Can we make a single macro that constructs the different views using closure syntax?